[SharedCache] Serialize `SharedCache::m_symbolInfos`

WeiN76LQh opened the pull request:
`SharedCache::m_symbolInfos` isn't being serialized but there is an attempt to deserialize it. This commit adds in the code to serialize it.

I slightly modified the format because I didn't really understand how it was expected to be serialized based on the deserialization code. The deserialization code looked wrong to me but its likely a misunderstanding on my part. I kept it similar to how `m_exportInfos` is serialized.

Commit:
[SharedCache] Serialize `SharedCache::m_symbolInfos`

`SharedCache::m_symbolInfos` isn't being serialized but there is an attempt to deserialize it. This commit adds in the code to serialize it.

I slightly modified the format because I didn't really understand how it was expected to be serialized based on the deserialization code. The deserialization code looked wrong to me but its likely a misunderstanding on my part. I kept it similar to how `m_exportInfos` is serialized.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/Vector35/binaryninja-api?pullRequest=6174) <br/>All committers have signed the CLA.

0cyn commented:
Hi! I'll be taking a look at this PR in conjunction with https://github.com/Vector35/binaryninja-api/pull/6210, seeing as removing symbol info serialization entirely in favor of using User Symbols may be a better approach.

0cyn committed:
Update metadata version to prevent v2 loader incorrectly trying to load v3 symbol list

We should think through a better way of handling upgrades, as v3 can load v2 dbs just fine, but there is no clean way to upgrade the info currently. We need SharedCache ser/des functions in DSCView.cpp
